src/System.Private.CoreLib/src/System/Activator.cs

```diff
@@ -77,7 +78,26 @@ public static T CreateInstance<T>()
         }
 
         [Intrinsic]
-        private extern static T CreateInstanceIntrinsic<T>();
+        private static T CreateInstanceIntrinsic<T>()
+        {
+            // Fallback implementation for codegens that don't support this intrinsic.
+            // This uses the type loader and doesn't have the kind of guarantees about it always working
+            // as the intrinsic expansion has. Also, it's slower.
+
+            EETypePtr eetype = EETypePtr.EETypePtrOf<T>();
+
+            // The default(T) check can be evaluated statically and will result in the body of this method
+            // becoming empty for valuetype Ts. We still need a dynamic IsNullable check to cover Nullables though.
+            // This will obviously need work once we start supporting default valuetype constructors.
```

Comment:
(I track that in #2370 - I added a note about `Activator.CreateInstance<T>`, but this will be mostly about declaring that we support the feature at some point and then cleaning out the bug tail.)